adapters/rainbowkit-connector/src/index.ts

```diff
@@ -1,143 +1,10 @@
-    super.disconnect();
-    const blocto = await this.getBlocto();
-    if (!blocto) throw new ConnectorNotFoundError();
-    await blocto?.ethereum?.request({ method: 'wallet_disconnect' });
-  }
-  async getBlocto(): Promise<any> {
-    if (!this.blocto) {
-      const [defaultChain] = this.#chains ?? [];
-      this.blocto = new BloctoSDK({
-        ethereum: {
-          chainId: defaultChain.id,
-          rpc: defaultChain.rpcUrls?.default?.http[0],
-        },
-      });
-    }
-    return this.blocto;
-  }
-  async getProvider(): Promise<any> {
-    const blocto = this.blocto ?? (await this.getBlocto());
-    if (!this.provider) {
-      this.provider = blocto.ethereum;
-    }
-    return this.provider;
-  }
-
-  async getAccount(): Promise<any> {
-    const blocto = await this.getBlocto();
-    if (!blocto) throw new ConnectorNotFoundError();
-    const accounts = await blocto?.ethereum?.request({
-      method: 'eth_requestAccounts',
-    });
-    if (!accounts?.length) throw new ConnectorNotFoundError();
-    return accounts[0];
-  }
-  async getChainId(): Promise<number> {
-    const provider = await this.getProvider();
-    if (!provider) throw new ConnectorNotFoundError();
-    return provider
-      .request({ method: 'eth_chainId' })
-      .then((chainId: string): number => parseInt(chainId));
-  }
-  /**
-   * @see https://docs.blocto.app/blocto-sdk/javascript-sdk/evm-sdk/switch-ethereum-chain
-   */
-  async switchChain(chainId: number): Promise<Chain> {
-    const defaultChainData = {
-      id: chainId,
-      name: `Chain ${chainId}`,
-      network: `${chainId}`,
-      nativeCurrency: { name: 'Ether', decimals: 18, symbol: 'ETH' },
-      rpcUrls: { default: { http: [''] }, public: { http: [''] } },
-    };
-    const chain =
-      this.#chains.find((x) => x.id === chainId) ?? defaultChainData;
-    const provider = await this.getProvider();
-    if (!provider) throw new ConnectorNotFoundError();
-    try {
-      const providerRpcurl =
-        provider._blocto.switchableNetwork[chainId]?.rpc_url;
-      const chainUrl = chain.rpcUrls?.default?.http[0];
-      if (providerRpcurl !== chainUrl) {
-        await provider.request({
-          method: 'wallet_addEthereumChain',
-          params: [
-            {
-              chainId,
-              rpcUrls: [chainUrl],
-            },
-          ],
-        });
-      }
-      await provider.request({
-        method: 'wallet_switchEthereumChain',
-        params: [{ chainId }],
-      });
-      return chain;
-    } catch (error: any) {
-      const chain = this.chains.find((x) => x.id === chainId);
-      if (!chain) {
-        throw new ConnectorNotFoundError('chain not found');
-      }
-      if (this.isUserRejectedRequestError(error)) {
-        throw new UserRejectedRequestError(error);
-      }
-      throw new SwitchChainError(error);
-    }
-  }
-}
-
 export const bloctoWallet = ({ chains }: BloctoWalletOptions): Wallet => ({
```

Comment:
Would you add a second paramter for passing `appId` and make it the second parameter of `BloctoConnector`'s `options` input?